src/sagas.js

```diff
@@ -295,19 +402,106 @@ function* schedulerPerMinute (action) {
+  const result = yield callExternalAction('addNewVm', Api.addNewVm, action)
+  if (result.error) {
+    let msg = (result.error.responseJSON && result.error.responseJSON.detail) || ''
+    yield put(updateVmDialogErrorMessage(msg.replace(/^\[|\]$/mg, '')))
+  } else {
+    yield put(closeDetail())
+    yield put(getAllVms({ shallowFetch: false }))
+  }
+}
+
+function* editVm (action) {
+  yield put(updateVmDialogErrorMessage(''))
+  const result = yield callExternalAction('editVm', Api.editVm, action)
+  if (result.error) {
+    let msg = (result.error.responseJSON && result.error.responseJSON.detail) || ''
+    yield put(updateVmDialogErrorMessage(msg.replace(/^\[|\]$/mg, '')))
+  } else {
+    yield put(closeDetail())
+    yield put(getAllVms({ shallowFetch: false }))
+  }
+}
+
+function* editTemplate (action) {
+  yield put(updateEditTemplateErrorMessage(''))
+  const result = yield callExternalAction('editTemplate', Api.editTemplate, action)
+  if (result.error) {
+    let msg = (result.error.responseJSON && result.error.responseJSON.detail) || ''
+    yield put(updateEditTemplateErrorMessage(msg.replace(/^\[|\]$/mg, '')))
+  } else {
+    yield put(closeDetail())
+    yield put(getAllVms({ shallowFetch: false }))
+  }
+}
+
+function* fetchAllTemplates (action) {
+  const templates = yield callExternalAction('getAllTemplates', Api.getAllTemplates, action)
+
+  if (templates && templates['template']) {
+    const templatesInternal = templates.template.map(template => Api.templateToInternal({ template }))
+    yield put(addTemplates({ templates: templatesInternal }))
+    // update template in store for add vm dialog
+    const activeTemplate = Selectors.getTemplateById('00000000-0000-0000-0000-000000000000')
+    yield put(updateTemplate(activeTemplate))
+    yield put(updateVmMemory(activeTemplate.memory))
+    yield put(updateVmCpu(activeTemplate.cpu))
+    yield put(updateVmName(''))
+    // update template in store for edit template dialog
+    yield put(updateEditTemplate(activeTemplate))
+  }
+}
+
+function* fetchAllClusters (action) {
+  const clusters = yield callExternalAction('getAllClusters', Api.getAllClusters, action)
+
+  if (clusters && clusters['cluster']) {
+    const clustersInternal = clusters.cluster.map(cluster => Api.clusterToInternal({ cluster }))
+    yield put(addClusters({ clusters: clustersInternal }))
+    yield put(updateCluster(clustersInternal[0]))
+  }
+}
+
+function* fetchAllOS (action) {
+  const operatingSystems = yield callExternalAction('getAllOperatingSystems', Api.getAllOperatingSystems, action)
+
+  if (operatingSystems && operatingSystems['operating_system']) {
+    const operatingSystemsInternal = operatingSystems.operating_system.map(os => Api.OSToInternal({ os }))
+    yield put(addAllOS({ os: operatingSystemsInternal }))
+    const activeOperatingSystem = operatingSystemsInternal.find(os => os.name === 'other')
+    yield put(updateOperatingSystem(activeOperatingSystem))
```

Comment:
This flow seems weird to me, please describe what's happening here.
Do I understand correctly, it's "updating a copy" held by potentially opened dialog? If so, the flow should be redesigned. @matobet , what do you think?